File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int  mul = 1;
        int cnt = -1;
        int n = nums.size();

        for(int i = 0 ; i < n ; i++){
            if(cnt == -1 && nums[i] == 0){
                cnt = i;
                continue;
            }
            else if(nums[i] == 0){
                cnt = -2;
                break;
            }
            mul *= nums[i];

        }    

        vector<int> ans(n);
        for(int i = 0 ; i < n ; i++){
            if(cnt == -2){
                ans[i] = 0;
            }
            else if(cnt != -1){
                ans[i] = i != cnt ? 0 : mul;
            }
            else{
                ans[i] = mul/nums[i];
            }
        }

        return ans;
    }
};

File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        i = 0 
        j = len(nums)-1;

        while i < j:
            sum = nums[i] + nums[j]
            if sum == target:
                return [i+1,j+1]
            elif sum > target:
                j -= 1
            else:
                i += 1

        return [0 , 0]